docs/pages/blog/mui-x-v7-beta.md

```diff
@@ -0,0 +1,245 @@
+    <img src="/static/blog/mui-x-v7-beta/intro.png" alt="MUI X v7 beta release" width="1200" height="400" style="width: 100%; height: 100%; object-fit: cover; object-position: center;" />
+  </a>
+</div>
+
+It's the end of January, and we're ready to start rolling out some of the plans for 2024!
+We have new components, exciting features, and a lot of improvements for both developers and end-users.
+
+Starting [now](https://github.com/mui/mui-x/releases/tag/v7.0.0-beta.0), MUI X v7 is in beta.
+Most breaking changes are in place, and we shift our focus towards refining and expanding over the new version.
+
+## Table of contents
+
+- [Data Grid](#data-grid)
+  - [Sticky headers](#sticky-headers)
+  - [Server-side data source](#server-side-data-source)[<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')
+  - [Date Object Support in Filter Model](#date-object-support-in-filter-model)
+  - [New stable features](#new-stable-features)
+- [Tree View](#tree-view)
+  - [RichTreeView](#richtreeview-new-component)
+- [Charts](#charts)
+  - [Reference Line](#reference-line)
+- [Date Pickers](#date-pickers)
+  - [Support for date-fns v3](#support-for-date-fns-v3)
+  - [Date Time Range Picker](#date-time-range-picker)[<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')
+- [What's next](#whats-next)
+- [Migration guide](#migration-guide)
+- [How to get involved](#how-to-get-involved)
+
+## Data Grid
+
+We have introduced critical breaking changes designed not only to enhance the developer experience but also to bring the user experience to a new level.
+
+### Sticky headers
+
+The column headers and pinning implementations have been refactored for an improved user experience.
+You'll notice a more responsive Data Grid with smoother vertical scrolling and better screen reader support.
+The video below showcases some of these improvements.
+
+<video preload="metadata" style="margin-bottom: 10px;" autoplay muted loop playsinline width="690" height="417" controls>
+  <source src="/static/blog/mui-x-v7-beta/sticky-headers.mp4" type="video/mp4">
+</video>
+
+:::warning
+The new experience is far better, but the solution has a limitation: **column headers can no longer be transparent** because content scrolls behind them.
+
+If you're using our theming system, the DataGrid will adapt to your application seamlessly.
+
+Without the theming system, you may need to manually set an appropriate background color to maintain your design.
+:::
+
+### Server-side data source
+
+Integration with the server side has been a huge pain point for our community; we aim to enhance developer experience and productivity by introducing the `DataSource` interface.
+
+This new interface establishes more efficient data management and seamless interactions, allowing developers to build more responsive and robust applications that connect UI elements to backend services with less complexity.
```

Comment:
We should make it clear that this feature is not available yet, otherwise it could be misleading